h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -401,18 +409,33 @@ public void importFiles(String path, String pattern, ArrayList<String> files, Ar
       I[Value.HDFS].importFiles(path, pattern, files, keys, fails, dels);
     }
 
-    if(pattern != null && !pattern.isEmpty()) {
-      files.retainAll(matchPattern(path,files,pattern)); //New files ArrayList after matching pattern of choice
-      keys.retainAll(matchPattern(path,keys,pattern)); //New keys ArrayList after matching pattern of choice
-      //New fails ArrayList after matching pattern of choice. Only show failures that match pattern
-      if(!fails.isEmpty()) {
-        fails.retainAll(matchPattern(path, fails, pattern));
-      }
+    final String[] patterns = pattern != null && !pattern.isEmpty() ? 
```

Comment:
nit: Isn't this condition redundant with respect to the condition at line [415](https://github.com/h2oai/h2o-3/pull/5204/files#diff-658af709365770678f5a8b58440d20a396e79dd4c0c9f54398eee4b48aaefeedR415)?

Reply:
it is, its only purpose is to avoid copying the array in case the user-provided pattern is actually empty; but yes, is technically redundant